tap-agent/src/agent/sender_account.rs

```diff
@@ -129,6 +135,56 @@ impl State {
         self.sender_fee_tracker.update(allocation_id, result.value);
         Ok(())
     }
+
+    /// Deny the sender if the total fee is greater than the max-unnaggregated-fees-per-sender.
+    /// Will update [`State::deny`], as well as the denylist table in the database.
+    async fn maybe_deny(&mut self) {
+        if self.sender_fee_tracker.get_total_fee()
+            >= self.config.tap.max_unnaggregated_fees_per_sender.into()
+        {
```

Comment:
I'd rather use `add_to_denylist()` and `remove_from_denylist()` and use the condition where it's supposed to be used, we could also have a `should_deny() -> bool` and just update the condition. 

An alternative could also be something like `verify_and_update_denylist()` where the function could add to the denylist or remove depending on the conditions.

Reply:
> An alternative could also be something like verify_and_update_denylist() where the function could add to the denylist or remove depending on the conditions.

I prefer add to denylist before the RAV request, and remove after the RAV request, to remove any opportunity for circumvention (as explained in comments)

Reply:
>I prefer add to denylist before the RAV request, and remove after the RAV request, to remove any opportunity for circumvention (as explained in comments)

In that case, you could call it twice 😁 but it's just an alternative.

Reply:
Here's a refactor: https://github.com/graphprotocol/indexer-rs/pull/168/commits/f66879adfeacd7b283203bd5e4e55f283d5ad64d
Let me know if that's what you had in mind